gci/componentmodel.py

```diff
@@ -437,6 +452,15 @@ class Resource(Artifact, LabelMethodsMixin):
     labels: typing.List[Label] = dataclasses.field(default_factory=tuple)
     srcRefs: list[SourceReference] = dataclasses.field(default_factory=tuple)
 
+    def __post_init__(self):
+        if dataclasses.is_dataclass((access := self.access)):
+            return
+
+        if isinstance(access, dict):
+            if not 'type' in access:
+                raise ValueError('attribute `type` must be present')
+            self.access = AccessDict(access)
```

Comment:
Maybe keep dataclass frozen and instead set the value here using `object.__setattr__(self, 'access', AccessDict(access))`? That way, unwanted mutation would still be prevented